Core/Commands/CommandExtensions.cs

```diff
@@ -42,5 +44,15 @@ public static Command WithPerUserCooldown(this Command command, Duration duratio
                         : Task.FromResult(new CommandResult()))
             { Aliases = command.Aliases, Description = command.Description };
         }
+
+        public static Command WithRestrictedByUser(this Command command, Func<User, bool> canExecute)
```

Comment:
I'm wondering if it's better to do restrictions based on attributes instead of chaining them, for example
[OperatorOnly] or [ModeratorOnly]

Reply:
I don't know how I would do that without
- complicate the command execution signature, or
- relying on reflection to read those attributes and modify the execution accordingly, which relies on "magic" that the compiler wouldn't see if it suddenly didn't happen anymore